lib/misc/firestore.dart

```diff
@@ -0,0 +1,56 @@
+import 'package:cloud_firestore/cloud_firestore.dart';
+import 'package:latlong2/latlong.dart';
+import 'package:tg_proj/misc/auth.dart';
+
+class Firestore {
+  static final Firestore instance = Firestore._();
+
+  Firestore._();
+
+  final user = Auth.instance.currentUser;
+
+  Future<void> createDocOnRegister() async {
+    await FirebaseFirestore.instance.collection('users').doc(user!.uid).set({
+      'email': user!.email,
```

Comment:
Make sure email is not null beforehand, otherwise the app crashes. If you don't want to make an if statement here, put a check somewhere else in the app which would run before this could ever run